design/sds-design.md

```diff
@@ -0,0 +1,68 @@
+# Support Secret Discovery Service on Contour
+
+_Status_: Draft
+
+This document outlines what changes are needed on Contour to support SDS. [0]
+
+## Goals
+
+- Implement SDS server as a gRPS service
```

Comment:
I still don't have the full picture regarding how things should work end to end. For instance, in the certe rotation scenario, how does SDS management server we plan to add here, works with some existing cert management tool, to complete the whole flow.

Meanwhile, is it possible to add initial support to Contour so that it can enable SDS config in Envoy, but pointing to 3rd party SDS management server, rather than implementing in Contour codebase itself. For example, run Contour for LDS that points to Istio `node_agent` and `citadel` for providing SDS.

@bmadhavan @davecheney comments?